nntrainer/graph/network_graph.cpp

```diff
@@ -696,45 +696,64 @@ void NetworkGraph::inPlaceOptimize(Manager &manager) {
   }
 }
 
-void NetworkGraph::updateRunContext(std::shared_ptr<Manager> &manager,
-                                    const std::shared_ptr<LayerNode> &lnode) {
+std::vector<Var_Grad *>
+NetworkGraph::updateRunContext(std::shared_ptr<Manager> &manager,
+                               const std::shared_ptr<LayerNode> &lnode,
+                               const std::vector<Var_Grad *> &prev_inputs) {
   /**
    * using copy assignment allows setting run_context without adding more
    * interfaces
    */
   const GraphNode &gnode = *lnode.get();
   const InitLayerContext &init_context = lnode->getInitContext();
+  std::vector<Var_Grad *> inputs = prev_inputs;
+  if (inputs.empty())
+    inputs = manager->requestInputs(gnode, init_context.getInputDimensions());
+
```

Comment:
It looks like using A output as an input of B in A-->B situation. is it enough just set like this? or are you going to optimize this?

Reply:
> It looks like using A output as an input of B in A-->B situation. 

Yes, this marks sharing the Var_Grad wrappers. This does not include the optimization of the shared tensors. That is done by the manager.

> is it enough just set like this? or are you going to optimize this?

Yes, this is enough to automatically redirect the output of layer A to the input of layer B.